metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -600,6 +603,24 @@ def get_datasource_urn(
 
         return None
 
+    @staticmethod
+    def strip_template_expressions(raw_query) -> str:
+        """
+        Workarounds for metabase raw queries containing most commonly used template expressions:
+
+        - strip conditional expressions "[[ .... ]]"
+        - replace all {{ filter expressions }} with "1"
+
+        reference: https://www.metabase.com/docs/latest/questions/native-editor/sql-parameters
+        """
+
+        # drop [[ WHERE {{FILTER}} ]]
+        query_patched = re.sub(r"\[\[.+\]\]", r" ", raw_query)
+
+        # replace {{FILTER}} with 1
+        query_patched = re.sub(r"\{\{.+\}\}", r"1", query_patched)
```

Comment:
Small improvement idea (which I assume you already indicated in the issue comment): we could theoretically put `.+` into a matcher group and only replace it if it's a know parameter by matching it against `card_details["parameters"]`.

Not sure if it's worth the effort, though, as non-replaced parameters would only cause the parser to fail and ignore the lineage for this card and there's likely no harm in globally replacing everything.

Reply:
Yes, I was considering doing that and decided against it to avoid introducing additional complexity without any practical benefit so far I could see. I mean, other than being more "correct" and pedantic. 

The correct place to fix that is upstream, all we can do downstream is to apply workarounds and sanitization. An upstream "fix" should be rather an feature to query the dashboard and card characteristics such as dependencies / database objects being used but that is not trivial and IMHO unlikely to be implemented any time soon.